### PR TITLE
[internal refactor] minor tweaking

### PR DESCRIPTION
**Pull request by bobzhang**
Happen to read this part of code, not clear why it's written in that style. Note Hashtbl.length is O(1)

### CODE REVIEW

File: typing/typetexp.ml [hunk 1]
```diff
@@ -423,14 +423,11 @@ and transl_type_aux env policy styp =
...
-              Hashtbl.iter (fun _ _ -> raise Exit) hfields;
+            if Hashtbl.length hfields <> 0 then 
+              name := None 
+            else   
               name := nm
-            with Exit ->
-              (* Unset it otherwise *)
-              name := None
-            end;
+            ;
```
Comment (nojb) [inline typing/typetexp.ml]: I would write it in one line: `name := if Hashtbl.length hfields = 0 then nm else None;`.

**Review by nojb — APPROVED** [review]
LGTM. I left a style suggestion, but feel free to ignore it. For reference this code was introduced in 9ea5edac9ab0b3860688583a2ff22a9e164be086

**Comment by dra27** [comment]
`Hashtbl.length` hasn't always been there (it was introduced in 3.08.0 in 2005) 🙂

**Comment by gasche** [comment]
It's hard to imagine how tough life was for our grand-parents...

File: typing/typetexp.ml (hunk 1 above)
Comment (bobzhang) [reply typing/typetexp.ml]: addressed

File: typing/typetexp.ml [hunk 2]
```diff
@@ -422,15 +422,8 @@ and transl_type_aux env policy styp =
...
-            begin try
-              (* Set name if there are no fields yet *)
-              Hashtbl.iter (fun _ _ -> raise Exit) hfields;
-              name := nm
-            with Exit ->
-              (* Unset it otherwise *)
-              name := None
-            end;
+            in            
+            name := if Hashtbl.length hfields <> 0 then  None else nm;
```
Comment (nojb) [inline typing/typetexp.ml]: I'm splitting hairs here, but there is an extra space before `None`.

**Comment by nojb** [comment]
I took the liberty of pushing the fix for the extra space and a check-typo failure directly to your branch.

It is now merged, thanks!

**Pull request merged**